programs/pluto/ikev2_ts.h

```diff
@@ -33,6 +33,20 @@ struct traffic_selector {
 	uint16_t startport;
 	uint16_t endport;
 	ip_range net;	/* for now, always happens to be a CIDR */
+
+	/**
+	 * `sec_ctx` is the security label (if any) that is associated with
+	 * this Traffic Selector.
+	 * 
+	 * There are 2 cases where `sec_ctx` is non-NULL:
+	 *
+	 *  1. If `ts_type` == `IKEv2_TS_SECLABEL`
+	 *
+	 *  2. If `ts_type` corresponds to an address range _and_ the security
```

Comment:
shouldn't it always be the second case? i.e TS_SECLABEL should always be in conjunction with other parameters (like the address range that is mandatory)

Reply:
The first case is used during input TS Payload processing. I have added several comments to clarify this. Please let me know if you think it needs more explanation.